pkg/kopialib/storage/storage_interface.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	kopiaerrors "github.com/kanisterio/kanister/pkg/kopia/errors"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/pkg/errors"
+)
+
+type StorageType string
+
+const (
+	TypeS3        StorageType = "S3"
+	TypeAzure     StorageType = "Azure"
+	TypeFileStore StorageType = "FileStore"
+	TypeGCP       StorageType = "GCP"
+)
+
+type Storage interface {
```

Comment:
Should this type be named something else, because if we access it from another package, we would have to use `storage.Storage`.

Reply:
I will rename it to provider